jattoabdul/narwhalorders
# [Refactor] Setup sidekiq for webhook jobs, tests and lint with rubocop

**What does this PR do?**
- [x] It add sidekiq for job queues and implements custom webhook controller.
- [x] It adds custom webhook routes
- [x] It adds listing with `rubocop`
- [x] it adds specs database cleaner and faker setup

**Description of Task to be completed?**
>- After the app is installed, the app should monitor for new orders. When a new order is received it should check to see if there are any products in the order that contains the word “Narwhal”. If there are, then it should add a record in the DB or somewhere. We’ll call these “Narwhal Orders”.

**How should this be manually tested?**
- Go to the app URL,
- Install the app
- Create a new order or Pay for an order
- Load the app in your Shopify admin and see the newly-created or paid-for order.

- Run in the development to test and see job creation in the sidekiq GUI via `${host}/sidekiq`

**Any background context you want to provide?**
- N/A

**What are the relevant issues?**
- N/A

**Screenshots (if appropriate)**
- N/A

**Questions:**
- N/A
